### ISSUE TITLE
Allow selector level 4 commas on the same line despite selector-list-comma-newline-after

### ISSUE THREAD

**Issue by ahstro** [stylelint/stylelint#910]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

When writing CSS that uses [Selectors level 4](https://drafts.csswg.org/selectors-4/), one is allowed to use multiple arguments in some selectors, e.g. `:not(:hover, :active)`. As you can see, this syntax uses a comma to separate the arguments, which triggers `selector-list-comma-newline-after` if it's set to `always`.

> Which rule, if any, is this issue related to?

 selector-list-comma-newline-after

> What CSS is needed to reproduce this issue?

```css
:matches(:hover, :focus) { }
```
or
```css
:not(:hover, :focus) { }
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
     "selector-list-comma-newline-after": "always"
  }
}
```

> Which version of stylelint are you using?

`5.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

From within Vim with [Syntastic](https://github.com/scrooloose/syntastic)

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to future CSS syntax, specifically [Selectors level 4](https://drafts.csswg.org/selectors-4/), which is currently available through the [cssnext](http://cssnext.io) PostCSS plugin.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
Expected newline after "," (selector-list-comma-newline-after)
```

**Comment by jeddy3** [comment]
Thanks for the detailed report :)

@davidtheclark The rule is specific to the `selector-list`s and so I think we can safely ignore commas within parentheses. Would `styleSearch()`'s `outsideFunctionalNotation` handle this scenerio or would a change to `stylesSearch()` be needed?

**Comment by davidtheclark** [comment]
> we can safely ignore commas within parentheses

I agree.

I'd have to look into the specifics more to figure out the right fix. I really have to get to work today, though :)

**Comment by davidtheclark** [comment]
@ahstro If you can file a PR with at least failing tests that illustrate the need, that would really help speed up this fix.
